src/main/java/jenkins/bouncycastle/api/BouncyCastlePlugin.java

```diff
@@ -0,0 +1,96 @@
+public class BouncyCastlePlugin extends Plugin {
+
+    private static final Logger LOG = Logger.getLogger(BouncyCastlePlugin.class.getName());
+
+    private static final boolean isActive;
+
+    static {
+        // BouncyCastle FIPS is installed in the JVM, we can not install over the top of it so do not try
+        Provider p = Security.getProvider("BCFIPS");
+        isActive = (p == null);
+
+        LOG.log(Level.INFO,
+                isActive ? "BouncyCastle Providers from BouncyCastle API plugin will be active" :
+                           "Detected the precence of the BouncyCastle FIPS provider, the regular BouncyCastle jars will not be avaialble.");
+    }
+
+    @Override
+    @SuppressRestrictedWarnings(jenkins.util.AntClassLoader.class) // we are messing with the classloader and it has not changed in many many years
+    public void start() throws Exception {
+        if (!isActive) {
+            // Alternative BouncyCastle is installed do no not insert these libraries
+            return;
+        }
+        // this is the hairy part.
+        // add the BouncyCastle APIs into the classpath for other plugins (and this plugin to use!)
+        /*
+         * Whilst plugins that have code may go boom before this with class loading issues, extensions (at this point)
+         * have not been discovered, so this would only affect people using the deprecated `Plugin` class (like we are!)
+         */
+        final List<String> bcJars = getResourceFiles();
+        AntClassLoader cl = (AntClassLoader) this.getWrapper().classLoader;
+
+        // grab the needed resources
+        for (String jar : bcJars) {
+            File jarFile = locateJar(jar);
+            LOG.log(Level.INFO, () -> "Inserting " + jar + " into bouncycastle-api plugin classpath");
+            cl.addPathComponent(jarFile);
+        }
+        SecurityProviderInitializer.addSecurityProvider();
+    }
+
+    public static boolean isActive() {
+        return isActive;
+    }
+
+    private static File locateJar(String name) throws IOException {
+        LOG.log(Level.FINE, "Attempting to locate BouncyCastle Jar {0}", name);
+        URL resourceURL = BouncyCastlePlugin.class.getResource("/bc_jars/" + name);
+        if (resourceURL == null) {
+            throw new IOException("Could not locate " + name + " in plugin resources");
+        }
+        try {
+            File f = new File(resourceURL.toURI());
+            LOG.log(Level.FINE, () -> "Located " + name + "  at " + f.getAbsolutePath());
+            return f;
+        } catch (URISyntaxException ex) {
+            throw new IOException("Could not locate file for " + name + " with URI " + resourceURL, ex);
+        }
+    }
+
+    private static List<String> getResourceFiles() throws IOException {
+        LOG.log(Level.FINE, "Searching for BouncyCastle Jars...");
+
+        try (InputStream in = BouncyCastlePlugin.class.getResourceAsStream("/bc_jars");
```

Comment:
seems this works on Windows but not on Linux.
Need to try looking for the wars in another way.